### ISSUE TITLE
Inquiry on Transposon Taxonomy of sequences

### ISSUE THREAD

**Issue by motto7** [DerKevinRiehl/transposon_classifier_rfsb#4]
Hi, 
I used your tool to annotate novel genome of echinoderm species, and i'm curious about your taxonomy classification taxonomy proposal. 

In your TransposonDB.fasta file, there are lots of Zator sequences in P-MITE db (about 140k). However, I couldn't find any guidance for allocating those MITE sequences as Zator (2/1/5). In P-MITE db, there's no mention about Zator sequences.

I would appreciate it if you could let me know how to divide those sequence into Zator.

Thank you.

Sincerely,

Jaehoon jung.

**Comment by DerKevinRiehl** [comment]
Hello Jaehoon Jung,
thanks for your interest in Transposon Ultimate and your question :-)

In our [paper ](https://academic.oup.com/nar/article/50/11/e64/6541023) we define Zator sequences as subcategory of TIR, and not MITE:
![grafik](https://user-images.githubusercontent.com/50080024/210090127-b1cbe786-1c9b-4290-9366-e83e5492233c.png)

As you mention correctly, we report in Table S4 that there are lots of Zators in PMiteDB.
![grafik](https://user-images.githubusercontent.com/50080024/210090426-3de54189-cb48-43be-addd-74c5186a2fcf.png)

_In P-MITE db, there's no mention about Zator sequences._
You are right, so when we were creating TransposonDB we used the labels provided by the database PMiteDB:

<html xmlns:o="urn:schemas-microsoft-com:office:office"
xmlns:x="urn:schemas-microsoft-com:office:excel"
xmlns="http://www.w3.org/TR/REC-html40">
<body link=blue vlink=purple>

Database | Database Taxonomy | Our Taxonomy
-- | -- | --
PMITEdb | Tc1/Mariner | 2.1.1
PMITEdb | Novosib | 2.1.18
PMITEdb | hAT | 2.1.2
PMITEdb | Mutator | 2.1.3
PMITEdb | P | 2.1.6
PMITEdb | PIF/Harbinger | 2.1.8
PMITEdb | CACTA | 2.1.9

</body>
</html>

However, the table S4 does not report the original labels of sequences as provided by the PMite-DB.
The PMiteDB does not follow our taxonomy, however if you convert their labels into our taxonomy (like CACTA is considered a TIR and not a MITE) and not their taxonomy, then we have lots of TIRs rather than MITEs. Besides, the numbers in the table were generated using our RFSB classifier.
This means we have lots of TIRs in PMITE-DB, and second the subcategories (like ZATOR) were predicted using RFSB.

I think the description of the figure is a little misleading in this case.

Besides, it can be confusing that PMITE-DB calls itself a MITE DB even though many of its labels, in other taxonomy schemes, refer for example to TIR rather than MITE transposons^^.

Hope this could help a little, Please let me know if this clarified your questions,
Best,
Kevin

**Comment by motto7** [comment]
I really really appreciate your kind reply.


As far as i know, and from the figure 1B of your paper, the difference between TIR and MITE is presence of transposase.

In your TransposonDB, however, mean length of Zator is even less than the MITE's... and it seems too low to contain any transposase....

When I use your program to newly constructed eukaryote genome, I found so many Zators ... and I could not find too little transposes. I think most of them should be MITE because it has TIR, DSD but no transposes.


if there is any other tool or pipeline to define TIR or MITE, please let me know.

Sorry for confusing, and really thank you for your fascinating tool.

**Comment by DerKevinRiehl** [comment]
Hello Jaehoon Jung,
great my answer could help a little :-).

I dont want to get philosophic here, but when creating the database we were differentiating between complete transposons and fragments of transposons.

Talking about "complete" transposons you are right, due to their length probably they are not able to be considered as complete transposons as they lack the full presence of all required consituents. However, using machine learning and relying on "k-mers" mainly, one could argue that the Zators we found might be old, deteriorated Zators that just resemble the "complete" transposons because of their k-mer signature. So what makes a transposon a transposon? hard to say^^

Lets put it that way: the software classifies transposons, even the deteriorated ones, based on their k-mer signature.
Any approach is based on assumptions, these are the assumptions for our tool.

Talking about tools that might help, I can recommend (if you want tools that solely annotate TIR and MITES) the ones marked red in following table from my masterthesis:
![grafik](https://user-images.githubusercontent.com/50080024/210134378-2675c3c0-2c14-4f41-9447-35831af170b1.png)

However, not all of these tools were integrated in transposonUltimate as I couldnt implement all of them in the conda package management system. But maybe this might be of use a little.

Best regards,
Kevin

**Comment by motto7** [comment]
I totally agree with you. Now I understand the things you intended.
I thought about it for a few months... and I found some slight clues from you.

I found that Zators what I found were correlated with heterozygosity increase (Even higher when active).. but there have been slight doubts about Zator signatures (with various TIRs and TSDs) on me.

Now I should deeply investigate on Zators what I found ,, still difficult though.

If you don't mind, i would like to discuss my results in the near future.
My paper using your tool is on review state, and i'm doing additional analysis now..

Thank you for your kind and fast reply.
 I hope you have a happy and healthy new year ^오^.


Sincerely,

Jaehoon Jung.
